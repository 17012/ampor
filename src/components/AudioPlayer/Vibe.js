import React from 'react'
import styled, { keyframes } from 'styled-components'

const pulse = keyframes`
  0%,100% {
    height: 40px;
  }
  50% {
    height: 100px;
  }
`

const Container = styled.div`
  & > div {
    display: inline-block;
    width: calc(100% / 20);
    height: 40px;
    opacity: 0.4;
    background-color: white;
    animation: ${pulse} 2s var(--number-count)
      cubic-bezier(0.28, 1.69, 0.82, -0.61) infinite;
  }
`

const Vibe = () => {
  return (
    <Container>
      {Array(20)
        .fill()
        .map((v, i) => (
          <div
            style={{ '--number-count': i * Math.random() * 100 + 'ms' }}
          ></div>
        ))}
    </Container>
  )
}

export default Vibe
