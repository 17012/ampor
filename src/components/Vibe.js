import React from 'react'
import styled, { keyframes } from 'styled-components'

const pulse = keyframes`
  0%,100% {
    height: 30px;
  }
  50% {
    height: 100px;
  }
`

const Container = styled.div`
  position: relative;

  & > div {
    position: absolute;
    bottom: 0;
    left: calc(100% / 20 * var(--number-count));
    width: calc(100% / 20);
    opacity: 0.4;
    background-color: white;
    animation: ${pulse} 1s var(--number-delay) ease-in infinite;
    animation-play-state: ${props => (props.playing ? 'running' : 'paused')};
  }
`

const Vibe = ({ playing }) => {
  return (
    <Container playing={playing}>
      {Array(20)
        .fill()
        .map((v, i) => (
          <div
            key={i}
            style={{
              height: ((i + 1) * Math.random() * 60) % 60,
              '--number-delay': (i + 1) * Math.random() * 60 + 'ms',
              '--number-count': i,
            }}
          ></div>
        ))}
    </Container>
  )
}

export default React.memo(Vibe)
