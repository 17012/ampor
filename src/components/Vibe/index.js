import Vibe from './Vibe'

export default Vibe
