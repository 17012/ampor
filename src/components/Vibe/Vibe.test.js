import React from 'react'
import Vibe from './Vibe'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<Vibe playing={true} />).toJSON()
  expect(tree).toMatchSnapshot()
})
