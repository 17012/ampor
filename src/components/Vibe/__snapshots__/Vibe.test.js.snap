// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm jDDlwE"
>
  <div
    style={
      Object {
        "--number-count": 0,
        "--number-delay": "17.796943913173035ms",
        "height": 36.13767018736932,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 1,
        "--number-delay": "111.82959066280782ms",
        "height": 25.151828831499294,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 2,
        "--number-delay": "43.16051289057211ms",
        "height": 53.163561737572124,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 3,
        "--number-delay": "46.16464938960913ms",
        "height": 30.77717567247322,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 4,
        "--number-delay": "115.4143777934439ms",
        "height": 40.597109529957606,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 5,
        "--number-delay": "287.3686715241455ms",
        "height": 11.028934938368536,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 6,
        "--number-delay": "24.60482206294703ms",
        "height": 13.960750739054689,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 7,
        "--number-delay": "470.22112534815994ms",
        "height": 23.617042864878357,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 8,
        "--number-delay": "426.91969624205706ms",
        "height": 32.09267151619245,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 9,
        "--number-delay": "260.3729587064195ms",
        "height": 6.845255187062406,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 10,
        "--number-delay": "639.0949433657314ms",
        "height": 45.47889650995887,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 11,
        "--number-delay": "536.6554929770938ms",
        "height": 16.47128442168382,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 12,
        "--number-delay": "345.7478709160624ms",
        "height": 3.5416766583881696,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 13,
        "--number-delay": "832.1590596724598ms",
        "height": 16.787426719396535,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 14,
        "--number-delay": "177.80281854341263ms",
        "height": 27.664820320275567,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 15,
        "--number-delay": "4.471244051646792ms",
        "height": 42.5955845147991,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 16,
        "--number-delay": "922.7968345276121ms",
        "height": 13.584342486393524,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 17,
        "--number-delay": "937.3031570270438ms",
        "height": 59.400852705953014,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 18,
        "--number-delay": "976.7175620921464ms",
        "height": 0.2894213002966808,
      }
    }
  />
  <div
    style={
      Object {
        "--number-count": 19,
        "--number-delay": "755.7606097680732ms",
        "height": 47.862064724020215,
      }
    }
  />
</div>
`;
